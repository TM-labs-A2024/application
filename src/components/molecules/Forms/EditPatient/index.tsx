import EditPatientForm from './EditPatient'

export default EditPatientForm
