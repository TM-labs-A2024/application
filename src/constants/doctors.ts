export const doctors = [
  {
    id: 1000000,
    specialities: [1],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 2000000,
    specialities: [2, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 3000000,
    specialities: [3, 5, 7],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 4000000,
    specialities: [4, 1],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 5000000,
    specialities: [5, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 6000000,
    specialities: [6, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 7000000,
    specialities: [7, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 8000000,
    specialities: [8, 9, 5, 1, 3, 4],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 9000000,
    specialities: [9, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 10000000,
    specialities: [10, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 11000000,
    specialities: [11, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 12000000,
    specialities: [12, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 13000000,
    specialities: [13, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 14000000,
    specialities: [14, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 15000000,
    specialities: [15, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 16000000,
    specialities: [16, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 17000000,
    specialities: [17, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 18000000,
    specialities: [18, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 19000000,
    specialities: [19, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 20000000,
    specialities: [20, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 21000000,
    specialities: [21, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 22000000,
    specialities: [22, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 23000000,
    specialities: [23, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 24000000,
    specialities: [24, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 25000000,
    specialities: [25, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 26000000,
    specialities: [26, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 27000000,
    specialities: [27, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 28000000,
    specialities: [28, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 29000000,
    specialities: [29, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 30000000,
    specialities: [30, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 31000000,
    specialities: [31, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 32000000,
    specialities: [32, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 33000000,
    specialities: [33, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 34000000,
    specialities: [34, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 35000000,
    specialities: [35, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 36000000,
    specialities: [36, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 37000000,
    specialities: [37, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 38000000,
    specialities: [38, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 39000000,
    specialities: [39, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 40000000,
    specialities: [40, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 41000000,
    specialities: [41, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 42000000,
    specialities: [42, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 43000000,
    specialities: [43, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 44000000,
    specialities: [44, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 45000000,
    specialities: [45, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 46000000,
    specialities: [46, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 47000000,
    specialities: [47, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 48000000,
    specialities: [48, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  }
]
