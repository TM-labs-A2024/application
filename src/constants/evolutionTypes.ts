export const evolutionTypes = [
  {
    id: '1',
    name: 'Evolución'
  },
  {
    id: '2',
    name: 'Hospitalización'
  },
  {
    id: '3',
    name: 'Alta'
  },
  {
    id: '4',
    name: 'Primera consulta'
  }
]
