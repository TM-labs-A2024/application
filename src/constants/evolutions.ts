export const evolution = {
  specialty: {
    description: '',
    id: '',
    name: ''
  },
  author: '',
  title: '',
  description: '',
  'content-format': '',
  created_at: '',
  payload: {
    type: {
      value: '',
      label: ''
    },
    reason: '',
    description: '',
    history: '',
    examination: '',
    summary: '',
    diagnostic: '',
    comments: ''
  }
}
