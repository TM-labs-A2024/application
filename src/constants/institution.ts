export const institution = {
  id: '3fa85f64-5717-4562-b3fc-2c963f66afa1',
  name: 'Clínica Tex Mex',
  credentials: '14578245',
  address: 'Av. Fernandez Peña, 147.',
  email: 'test@test.com',
  phone: '+582742214441'
}
