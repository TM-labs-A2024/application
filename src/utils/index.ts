import { formatDate } from './dates'
import { sendEmail, generateVerificationCode, setupEmailSending } from './email'
import { setupErrorNotification } from './notifications'
import { isWeb, isIOS, isAndroid, isMobile } from './platform'

export {
  isWeb,
  isIOS,
  isAndroid,
  isMobile,
  sendEmail,
  generateVerificationCode,
  setupEmailSending,
  setupErrorNotification,
  formatDate
}
