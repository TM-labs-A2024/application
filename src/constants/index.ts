import { doctors } from './doctors'
import { evolutionTypes } from './evolutionTypes'
import { institution } from './institution'
import { institutions } from './institutions'
import {
  ACCESS_REMOVED,
  FILTERS_APPLIED,
  ACCESS_REQUESTED,
  ATTACHMENT_DELETED,
  IMAGE_DELETED,
  ACCESS_DENIED,
  ACCESS_GRANTED,
  ACCESS_REMOVAL,
  GENERIC_NOTIFICATION,
  GENERIC_ERROR
} from './notifications'
import { nurse } from './nurses'
import { patient, sexOptions, statusOptions } from './patient'
import { patients } from './patients'
import { specialties, specialities } from './specialties'
import { specialtyData } from './specialtyData'

export {
  specialties,
  specialities,
  doctors,
  nurse,
  patient,
  sexOptions,
  statusOptions,
  specialtyData,
  evolutionTypes,
  patients,
  institution,
  institutions,
  ACCESS_REMOVED,
  FILTERS_APPLIED,
  ACCESS_REQUESTED,
  ATTACHMENT_DELETED,
  IMAGE_DELETED,
  ACCESS_DENIED,
  ACCESS_GRANTED,
  ACCESS_REMOVAL,
  GENERIC_NOTIFICATION,
  GENERIC_ERROR
}
