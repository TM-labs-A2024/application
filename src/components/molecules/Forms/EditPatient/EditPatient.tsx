import { ArrowBackIcon } from '@chakra-ui/icons'
import {
  Button,
  FormControl,
  FormErrorMessage,
  Input,
  Heading,
  InputLeftAddon,
  InputGroup,
  Stack,
  IconButton,
  Text
} from '@chakra-ui/react'
import { Patient } from '@src/types'
import { formatDate } from '@src/utils'
import { isIOS } from '@utils/index'
import { format } from 'date-fns'
import { useRouter } from 'next/navigation'
import React, { ReactElement } from 'react'
import { useForm, FieldErrors } from 'react-hook-form'

type FormData = {
  firstname: string
  lastname: string
  govId: string
  birthdate: string
  email: string
  phoneNumber: string
}

export default function EditPatientForm({
  context: { patient, onSubmit }
}: {
  context: { patient: Patient; onSubmit: (patient: Patient) => void }
}): ReactElement {
  // --- Hooks -----------------------------------------------------------------
  const {
    handleSubmit,
    formState: { errors, isSubmitting },
    register
  } = useForm<FormData>()

  const router = useRouter()
  // --- END: Hooks ------------------------------------------------------------

  // --- Local state -----------------------------------------------------------
  // --- END: Local state ------------------------------------------------------

  // --- Refs ------------------------------------------------------------------
  // --- END: Refs -------------------------------------------------------------

  // --- Redux -----------------------------------------------------------------
  // --- END: Redux ------------------------------------------------------------

  // --- Side effects ----------------------------------------------------------
  // --- END: Side effects -----------------------------------------------------

  // --- Data and handlers -----------------------------------------------------
  const onSubmitDetails = (data: FormData) => {
    onSubmit(data)
  }

  const verifyErrors = (errors: FieldErrors<FormData>) => Object.keys(errors).length > 0
  // --- END: Data and handlers ------------------------------------------------

  return (
    <div
      className={`flex h-screen w-screen flex-col p-8 lg:px-96 ${isIOS() ? 'pt-20' : 'pt-8'}`}
      data-testid="edit-patient-form"
    >
      <div className="mb-8 flex flex-row items-center justify-start">
        <IconButton
          size="xl"
          className="mr-4"
          aria-label="back"
          variant="link"
          icon={<ArrowBackIcon />}
          onClick={() => {
            router.push('/perfil')
          }}
        />
        <Text className="font-medium">Editar perfil</Text>
      </div>
      <form onSubmit={handleSubmit(onSubmitDetails)} className="w-full grow">
        <FormControl
          isInvalid={verifyErrors(errors)}
          className="relative flex h-full w-full flex-col justify-between"
        >
          <Heading as="h2" size="sm" mb={4} noOfLines={1}>
            Datos personales
          </Heading>
          <Stack className="h-3/4 overflow-scroll">
            <Input
              id="firstname"
              className="mt-2 min-h-10"
              defaultValue={patient.firstname}
              placeholder="Nombres"
              {...register('firstname', {
                required: 'Este campo es obligatorio'
              })}
            />
            <FormErrorMessage>{errors?.firstname && errors?.firstname?.message}</FormErrorMessage>
            <Input
              id="lastname"
              className="mt-2 min-h-10"
              defaultValue={patient.lastname}
              placeholder="Apellidos"
              {...register('lastname', {
                required: 'Este campo es obligatorio'
              })}
            />
            <FormErrorMessage>{errors?.lastname && errors?.lastname?.message}</FormErrorMessage>
            <Input
              id="id"
              className="mt-2 min-h-10"
              defaultValue={patient.govId}
              placeholder="Cédula"
              {...register('govId', {
                required: 'Este campo es obligatorio'
              })}
            />
            <FormErrorMessage>{errors?.govId && errors?.govId?.message}</FormErrorMessage>
            <Input
              id="birthdate"
              className="mt-2 min-h-10"
              defaultValue={format(new Date(formatDate(patient.birthdate)), 'yyyy-MM-dd')}
              type="date"
              placeholder="Fecha de nacimiento"
              {...register('birthdate', {
                required: 'Este campo es obligatorio'
              })}
            />
            <FormErrorMessage>{errors?.birthdate && errors?.birthdate?.message}</FormErrorMessage>
            <Input
              id="email"
              className="mt-2 min-h-10"
              defaultValue={patient.email}
              placeholder="Correo electrónico"
              {...register('email', {
                required: 'Este campo es obligatorio'
              })}
            />
            <FormErrorMessage>{errors?.email && errors?.email?.message}</FormErrorMessage>
            <InputGroup className="mt-2">
              <InputLeftAddon>+58</InputLeftAddon>
              <Input
                id="phone"
                className="min-h-10"
                defaultValue={patient.phoneNumber}
                type="tel"
                placeholder="Teléfono"
                {...register('phoneNumber', {
                  required: 'Este campo es obligatorio'
                })}
              />
            </InputGroup>
            <FormErrorMessage>
              {errors?.phoneNumber && errors?.phoneNumber?.message}
            </FormErrorMessage>
          </Stack>
          <Button isLoading={isSubmitting} type="submit" className="absolute top-0">
            Guardar
          </Button>
        </FormControl>
      </form>
    </div>
  )
}
