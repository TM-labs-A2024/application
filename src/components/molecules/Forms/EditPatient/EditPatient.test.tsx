import { patient } from '@src/constants'
import { render, screen } from '@test/utils'
import { format } from 'date-fns'
import 'next/navigation'

import EditPatient from './EditPatient'
import '@testing-library/jest-dom'

jest.mock('next/navigation')

describe('Molecules > Forms > EditPatient test', () => {
  const context = {
    patient,
    onSubmit: jest.fn()
  }
  test('The component requires the fields: Firstname, lastname, govId, birthdate, email and phoneNumber', () => {
    render(<EditPatient context={context} />)

    expect(screen.getByPlaceholderText('Nombres')).toBeInTheDocument()
    expect(screen.getByPlaceholderText('Apellidos')).toBeInTheDocument()
    expect(screen.getByPlaceholderText('Cédula')).toBeInTheDocument()
    expect(screen.getByPlaceholderText('Fecha de nacimiento')).toBeInTheDocument()
    expect(screen.getByPlaceholderText('Correo electrónico')).toBeInTheDocument()
    expect(screen.getByPlaceholderText('Teléfono')).toBeInTheDocument()
  })

  test("The component shows the placeholder text with the patient's data", () => {
    render(<EditPatient context={context} />)

    const formatedDate = format(new Date(patient.birthdate), 'yyyy-MM-dd')

    expect(screen.getByDisplayValue(patient.firstname)).toBeInTheDocument()
    expect(screen.getByDisplayValue(patient.lastname)).toBeInTheDocument()
    expect(screen.getByDisplayValue(patient.govId)).toBeInTheDocument()
    expect(screen.getByDisplayValue(formatedDate)).toBeInTheDocument()
    expect(screen.getByDisplayValue(patient.email)).toBeInTheDocument()
    expect(screen.getByDisplayValue(patient.phoneNumber)).toBeInTheDocument()
  })

  test('Matches the snapshot', () => {
    render(<EditPatient context={context} />)

    expect(screen.getByTestId('edit-patient-form')).toMatchSnapshot()
  })
})
