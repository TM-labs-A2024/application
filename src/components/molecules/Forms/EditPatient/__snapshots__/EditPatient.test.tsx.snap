// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules > Forms > EditPatient test Matches the snapshot 1`] = `
<div
  class="flex h-screen w-screen flex-col p-8 lg:px-96 pt-8"
  data-testid="edit-patient-form"
>
  <div
    class="mb-8 flex flex-row items-center justify-start"
  >
    <button
      aria-label="back"
      class="chakra-button mr-4 css-84vx7q"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="chakra-icon css-onkibi"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          fill="currentColor"
        />
      </svg>
    </button>
    <p
      class="chakra-text font-medium css-0"
    >
      Editar perfil
    </p>
  </div>
  <form
    class="w-full grow"
  >
    <div
      class="chakra-form-control relative flex h-full w-full flex-col justify-between css-1kxonj9"
      role="group"
    >
      <h2
        class="chakra-heading css-89pyn4"
      >
        Datos personales
      </h2>
      <div
        class="chakra-stack h-3/4 overflow-scroll css-8g8ihq"
      >
        <input
          class="chakra-input mt-2 min-h-10 css-1cjy4zv"
          id="firstname"
          name="firstname"
          placeholder="Nombres"
          value="José Feliciano"
        />
        <input
          class="chakra-input mt-2 min-h-10 css-1cjy4zv"
          id="lastname"
          name="lastname"
          placeholder="Apellidos"
          value="Gutierrez García"
        />
        <input
          class="chakra-input mt-2 min-h-10 css-1cjy4zv"
          id="id"
          name="govId"
          placeholder="Cédula"
          value="E99885561"
        />
        <input
          class="chakra-input mt-2 min-h-10 css-1cjy4zv"
          id="birthdate"
          name="birthdate"
          placeholder="Fecha de nacimiento"
          type="date"
          value="1999-11-11"
        />
        <input
          class="chakra-input mt-2 min-h-10 css-1cjy4zv"
          id="email"
          name="email"
          placeholder="Correo electrónico"
          value="user@example.com"
        />
        <div
          class="chakra-input__group mt-2 css-1y0e7gb"
          data-group="true"
        >
          <div
            class="chakra-input__left-addon css-127xmqf"
          >
            +58
          </div>
          <input
            class="chakra-input min-h-10 css-10wjqo2"
            id="phone"
            name="phoneNumber"
            placeholder="Teléfono"
            type="tel"
            value="+584127659814"
          />
        </div>
      </div>
      <button
        class="chakra-button absolute top-0 css-ez23ye"
        type="submit"
      >
        Guardar
      </button>
    </div>
  </form>
</div>
`
